src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,29 @@
+package duke;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+
+    protected LocalDate by;
```

Comment:
Perhaps you might want to consider a more descriptive variable name like `byDate`, so that readers can easily differentiate between this variable and the other `by` parameter in your constructor